pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -712,19 +714,26 @@ Result ConsumerImpl::getConsumerStats(BrokerConsumerStats& brokerConsumerStats,
             LOG_DEBUG(getName() <<
                     " Sending ConsumerStats Command for Consumer - " << getConsumerId() << ", requestId - "<<requestId);
 
-            BrokerConsumerStats consumerStats;
-            Result res = cnx->newConsumerStats(topic_, subscription_, consumerId_, requestId).get(consumerStats);
-            if (res == ResultOk) {
-                brokerConsumerStats = brokerConsumerStats_ = consumerStats;
-            }
-            return res;
+            cnx->newConsumerStats(topic_, subscription_, consumerId_, requestId).addListener(
+                    boost::bind(&ConsumerImpl::brokerConsumerStatsListener, shared_from_this(), _1, _2, callback));
+            return;
         } else {
             LOG_ERROR(getName() << " Operation not supported since server protobuf version " << cnx->getServerProtocolVersion() << " is older than proto::v7");
-            return ResultOperationNotSupported;
+            return callback(ResultUnsupportedVersionError, BrokerConsumerStats());
         }
     }
     LOG_ERROR(getName() << " Client Connection not ready for Consumer");
-    return ResultNotConnected;
+    return callback(ResultNotConnected, BrokerConsumerStats());
+}
+
+void ConsumerImpl::brokerConsumerStatsListener(Result res, const BrokerConsumerStats brokerConsumerStats
+        , BrokerConsumerStatsCallback callback) {
+
+    if (res == ResultOk) {
+        Lock lock(mutex_);
```

Comment:
yes, to protect brokerConsumerStats_ in case getConsumerStats is called using multiple threads using the same consumer objects.